doc/developer/guide-changes.md

```diff
@@ -0,0 +1,699 @@
+    will be able to understand it more easily and more quickly.
+
+  * **Get better reviews.** The better your reviewer understands the
+    PR, the better a review you'll get. A reviewer is more able to
+    offer API design suggestions and determine whether the test coverage
+    is complete when the amount of code that is changed is small.
+
+  * **Have fewer merge conflicts.** The more code a PR touches, the
+    more likely it is to rot when other PRs land first.
+
+  * **Are easier to revert.** If a problem is discovered with a PR down the
+    line, it is much easier to revert if it is small and target.
+
+Exceptions:
+
+  * **Generated code.** When generated code is checked into the repository,
+    changes to the generated code can often result in hundreds or thousands
+    of lines of changes. Those changes don't need to count against the limit,
+    provided the reason for the change is clear (e.g., updating the version of
+    the dependency that generates the code).
+
+    Where possible, though, we avoid checking in generated code to the
+    repository
+
+  * **Declarative tests.** Declarative tests, like sqllogictest, testdrive,
+    and datadriven, are in general much easier to review than code. A PR that
+    changes 1000 lines of code is probably acceptable if 750 of the changed
+    lines results are new test cases.
+
+  * **Prose.** Written text is less dense than code. The appropriate maximum
+    size for a PR that adds only prose (e.g., a PR adding the documentation for
+    a new feature) is probably about 5,000 _words_. Reliably measuring prose in
+    number of lines is difficult as it depends heavily on formatting
+    details, like where lines are wrapped.
+
+This sentiment from the Google guide is particularly important to keep in mind:
+
+> Keep in mind that although you have been intimately involved with your code
+> from the moment you started to write it, the reviewer often has no context.
+> What seems like an acceptably-sized PR to you might be overwhelming to your
+> reviewer. When in doubt, write PRs that are smaller than you think you need to
+> write. Reviewers rarely complain about getting PRs that are too small.
+
+If you're finding it hard to explain a particular PR in writing, especially for
+large PRs, consider scheduling a brief meeting with your reviewer to walk them
+through the PR. Similarly, as a reviewer, if you have trouble understanding a
+PR, consider scheduling a meeting with your reviewee, rather than continuing to
+communicate by text. The downside is that your discussions won't be recorded for
+posterity, but the increased bandwidth of verbal communication can often make
+the tradeoff worthwhile.
+
+Some PRs, especially large refactors, are nearly impossible to split into
+pieces. If you find yourself in this situation, have a conversation with your
+reviewers ahead of time to get their buy in.
+
+### Early feedback
```

Comment:
I think it would be nice to also ask that engineers self-review their code (however they want) before submitting it for review by others? I've found that this often catches small things that would otherwise be needless cognitive overhead for the reviewer

Reply:
Good idea! Added a section called "polish requirements" that covers this.

Reply:
kind of in a similar vein -- i personally love when people call out the parts of their changes they feel most unsure about in the pr description - i know opinion is a little split on that. If you think its worth doing, might be nice to also add that to the guide?

Reply:
Sure, done!